src/pretix/presale/checkoutflow.py

```diff
@@ -301,15 +318,17 @@ def post(self, request):
                 self.cart_session['customer'] = request.customer.pk
                 self.cart_session['customer_cart_tied_to_login'] = True
                 return redirect(self.get_next_url(request))
-            elif self.login_form.is_valid():
+            elif "login-sso-data" in self.request.POST and self._handle_sso_login():
+                return redirect(self.get_next_url(request))
+            elif self.login_form.is_valid() and self.event.settings.customer_accounts_native:
```

Comment:
Tiny optimization, but check `customer_accounts_native` first to skip `is_valid` if it is not needed?

```suggestion
            elif self.event.settings.customer_accounts_native and self.login_form.is_valid():
```